src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+                    var tbInputSearch = (<HTMLInputElement>document.getElementById("fbSearchInput"));
+
+                    if(tbInputSearch === null)
+                    {
+                        var allByClassTest = document.querySelectorAll('div.bp3-transition-container.bp3-popover-enter-done div.bp3-popover-content > div > div');
+                        var testFilterDiv = (<HTMLElement>allByClassTest[0]);
+                        if(typeof testFilterDiv !== 'undefined' && testFilterDiv !== null)
+                        {
+                            var newDivLine = document.createElement('div');
+                                newDivLine.className = 'rm-line';
+                                testFilterDiv.prepend(newDivLine);
+
+                            var newDiv = document.createElement('div');
+                                newDiv.id = 'filterBoxSearch';
+                                newDiv.style.cssText = 'display:flex;padding:4px';
+                                testFilterDiv.prepend(newDiv);
+
+                            var newLabel = document.createElement('strong');
+                                newLabel.innerText = 'Search';
+                                newLabel.style.cssText = 'margin-right:10px';
+                                newDiv.appendChild(newLabel);
+
+                            var newInput = document.createElement('input');
+                                newInput.value = '';
+                                newInput.id = 'fbSearchInput';
+                                newInput.name = 'fbSearchInput';
+                                newInput.style.cssText = 'width:200px;display:flex';
+                                newDiv.appendChild(newInput);
+                                newInput.focus();
+
+                            newInput.addEventListener("input", newInputClick);
+                            return;
+                        }else{if(debugMode == 1){console.log('Filter box was not loaded in time!')}}
+                    }
+                    waitCtr++;
+                    if(debugMode == 1){console.log('*** WAIT LOOP: ',waitCtr)}
+                    if(waitCtr >= 12){return;}
+                    waitForFilter();
+                }, 50)
+            }
+
+            waitForFilter();
+        }
+    }
+}
+
+function newInputClick()
+{
+    var inputTxtVal = (<HTMLInputElement>document.getElementById("fbSearchInput")).value.toString();
+    if(debugMode == 1){console.log(inputTxtVal);}
+
+    //Get filter box (only works when opened)
+    var allByClass2 = document.querySelectorAll('.bp3-overlay.bp3-overlay-open.bp3-overlay-inline');
+    var filterBox = allByClass2[0];
+    if(typeof filterBox !== "undefined")
+    {
+        var allFilterButtons = filterBox.querySelectorAll('div:not(.flex-h-box) > div > button.bp3-button');
+
+        for(var i = 0; i < allFilterButtons.length; i++)
+        {
+            var curElement = allFilterButtons.item(i);
```

Comment:
convert the type of the element here so you don't have to do it inline each time (like `const curElement = allFilterButtons.item(i) as HTMLElement;`)